### PR TITLE
[MI-2613]: Fixed resetting error message on Link modal, showing descr…

### PR DESCRIPTION
**Pull request by avas27JTG**
…iptive error messages on UI and handling modals if account is not connected

### CODE REVIEW

**Review by raghavaggarwal2308 — APPROVED** [review]

File: webapp/src/pluginConstants/messages.ts [hunk 1]
```diff
@@ -15,4 +15,6 @@ export const error = {
     notAccessibleError: 'Looks like this project/organization does not exist or you do not have permissions to access it',
     adminAccessErrorForUnlinking: 'You do not have sufficient permissions to delete subscriptions for this project but you can still unlink the project',
     projectAlreadyLinkedError: 'This project is already linked.',
+    errorExpectedForOauthNotEnabled: 'failed to link Project: status: 401 Unauthorized',
+    errorMessageOauthNotEnabled: 'Looks like "third-party application access via OAuth" setting is not enabled for the organization or you do not have sufficient permissions to access it',
```
Comment (ayusht2810) [inline webapp/src/pluginConstants/messages.ts:19]:
```suggestion
    errorExpectedForOAuthNotEnabled: 'failed to link Project: status: 401 Unauthorized',
    errorMessageOAuthNotEnabled: 'Looks like "third-party application access via OAuth" setting is not enabled for the organization or you do not have sufficient permissions to access it',
```

**Review by ayusht2810 — APPROVED** [review]

File: server/plugin/utils.go [hunk 2]
```diff
@@ -454,7 +454,8 @@ func (p *Plugin) UpdatePipelineRunApprovalPost(approvalSteps []*serializers.Appr
 }
 
 func (p *Plugin) deleteSubscription(subscription *serializers.SubscriptionDetails, mattermostUserID string) (int, error) {
-	if statusCode, err := p.Client.DeleteSubscription(subscription.OrganizationName, subscription.SubscriptionID, mattermostUserID); err != nil {
+	// On deletion if a subscription is not found on the Azure DevOps portal then delete it from the Mattermost's KV store
+	if statusCode, err := p.Client.DeleteSubscription(subscription.OrganizationName, subscription.SubscriptionID, mattermostUserID); statusCode != http.StatusNotFound && err != nil {
```
Comment (manojmalik20) [inline server/plugin/utils.go:458]:
```suggestion
	if statusCode, err := p.Client.DeleteSubscription(subscription.OrganizationName, subscription.SubscriptionID, mattermostUserID); err != nil && statusCode != http.StatusNotFound {
```
Should we do this? Is there any case where status code is 404 and there could still be error?

File: webapp/src/utils/errorHandling.ts [hunk 3]
```diff
@@ -17,13 +17,19 @@ const getErrorMessage = (
         if (errorState?.status === 403 && errorState?.data.Error.includes(pluginConstants.messages.error.accessDenied)) {
             return pluginConstants.messages.error.adminAccessError;
         }
-        return pluginConstants.messages.error.generic;
+        if (errorState?.status === 500 && errorState?.data.Error.includes(pluginConstants.messages.error.failedToGetSubscriptions)) {
```
[the remark this answers is not in the document]
Comment (avas27JTG) [reply webapp/src/utils/errorHandling.ts:20]: we need to show a custom message only when both `status=500` and error message = `pluginConstants.messages.error.failedToGetSubscriptions` are present and for the rest errors with `500` we are showing the message coming from AZD API, changed this for below as well

File: server/plugin/utils.go (hunk 2 above)
Comment (avas27JTG) [reply server/plugin/utils.go:458]: Don't think so, status 404 should have only 1 use-case that a resource is not found so irrespective of error if a subscription is not found on AZD then we should remove it from MM as well

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**